### PR TITLE
Init core Nix functionality

### PR DESCRIPTION
**Pull request by zeapoz**
- Development environment changes:
  - Rust versions are now pinned like everything else.
  - Default devShell now ships stable Rust with nightly `fmt`, without relying on `rustup`.
  - CI-scrips can now be run individually or as a suite, e.g. `ci-fmt` will only run the `fmt` check while `ci-all` will run everything.

- Nix changes:
  - Introduces a `lib.nix` file containing useful functions for converting attribute sets to executable scripts.
  - Introduces a Nix template that can be instantiated in a new project with `nix flake init -t github:runziggurat/ziggurat-core`.
  - The flake now also outputs `buildInputs` and `scripts` which can be used and overridden across projects.

### CODE REVIEW

**Review by Rqnsom — APPROVED** [review]
I can't provide the most valuable feedback here, but it looks good - I can understand the context more this time!
EDIT: but wait for one more approval, I saw @asmie is also checking this one

**Review by asmie — APPROVED** [review]
Looks good. 👍 
Didn't check hashes correctness 😄

**Pull request merged**
